packages/ui/src/working-groups/components/RolesList.tsx

```diff
@@ -0,0 +1,83 @@
+import BN from 'bn.js'
+import React from 'react'
+import styled from 'styled-components'
+
+import { BadgeViolet } from '@/common/components/BadgeViolet'
+import { ButtonGhost } from '@/common/components/buttons'
+import { KebabMenuIcon } from '@/common/components/icons'
+import { List, ListItem } from '@/common/components/List'
+import { TextInlineBig, TokenValue } from '@/common/components/typography'
+import { WorkerWithDetails } from '@/working-groups/types'
+
+import { workerRoleTitle } from '../helpers'
+
+import {
+  OACItemInfo,
+  OACItemInfoTop,
+  OACItemSummary,
+  OACItemTitle,
+  OACSubscriptionWide,
+  OACWrap,
+  OpenItemSummaryColumn,
+} from './OpeningAndApplicationsComponents/OACStyledComponents'
+
+interface Props {
+  workers: WorkerWithDetails[]
+}
+
+export const RolesList = ({ workers }: Props) => (
+  <List>
+    {workers.map((worker) => (
+      <ListItem key={worker.group.name + worker.membership.id}>
+        <RolesListItem worker={worker} />
+      </ListItem>
+    ))}
+  </List>
+)
+
+const RolesListItem = ({ worker }: { worker: WorkerWithDetails }) => {
+  return (
+    <OACWrap>
+      <OACItemInfo>
+        <OACItemInfoTop>
+          <BadgeViolet>{worker.group.name}</BadgeViolet>
+          {worker.isLeader && <BadgeViolet>LEADER</BadgeViolet>}
+        </OACItemInfoTop>
+        <Title>{workerRoleTitle(worker)}</Title>
+      </OACItemInfo>
+      <OACItemSummary>
+        <OpenItemSummaryColumn>
+          <TextInlineBig>
+            <TokenValue value={new BN(worker.rewardPerBlock)} />
+          </TextInlineBig>
+          <OACSubscriptionWide>Reward per 3600 block.</OACSubscriptionWide>
+        </OpenItemSummaryColumn>
+        <OpenItemSummaryColumn>
+          <TextInlineBig>
+            <TokenValue value={new BN(worker.earnedTotal)} />
+          </TextInlineBig>
+          <OACSubscriptionWide>Earned total</OACSubscriptionWide>
+        </OpenItemSummaryColumn>
+        <OpenItemSummaryColumn>
+          <TextInlineBig>
+            <TokenValue value={new BN(2396000)} />
+          </TextInlineBig>
+          <OACSubscriptionWide>Next payment in</OACSubscriptionWide>
+        </OpenItemSummaryColumn>
+        <OpenItemSummaryColumn>
+          <TextInlineBig>
+            <TokenValue value={new BN(worker.stake)} />
+          </TextInlineBig>
+          <OACSubscriptionWide>Staked</OACSubscriptionWide>
+        </OpenItemSummaryColumn>
+      </OACItemSummary>
+      <ButtonGhost square size="medium">
+        <KebabMenuIcon />
+      </ButtonGhost>
+    </OACWrap>
```

Comment:
I wonder what to do with this one. The components' names suggest that those are for "application and openings" components. However, it looks like we can re-use them elsewhere.

I think that we need some better abstraction for this.

ps.: I'll keep in mind that, for now it is OK.

Reply:
Probably this should be some component in common, because it appears in a lot of places